File: Tests/c/eight.c
#include <stdio.h>
#include <stdlib.h>
#include <stdbool.h>
#include <assert.h>

typedef struct {
    int id;
    bool is_visible;
    char *label;
} GraphicElement;

GraphicElement* create_element(int id, char *label) {
    GraphicElement *el = (GraphicElement*)malloc(sizeof(GraphicElement));
    el->id = id;
    el->label = label;
    return el;
}

void render(GraphicElement *el) {
    if (el->is_visible) {
        printf("Zimeju elementu %d: %s\n", el->id, el->label);
    }
}

int main() {
    GraphicElement *btn = create_element(1, "Saglabat");

    render(btn);

    free(btn);
    printf("Scenārijs 8 darbojas veiksmīgi.\n");
    return 0;
}

File: Tests/python/eight.py
import uuid

class GraphicElement:

	def __init__(self):
		self.is_visible = True
		self.z_index = 0
		self._id = uuid.uuid4()

	def render(self):
		if self.is_visible:
			return f"Rendēju elementu {self._id} līmenī {self.z_index}"
		return ""


class ButtonWidget(GraphicElement):


	def __init__(self, label: str):
		self.label = label
		self.is_clicked = False




	def click(self):
		self.is_clicked = True
		print(f"Poga '{self.label}' noklikšķināta!")


if __name__ == "__main__":
	btn = ButtonWidget("Saglabāt")

	render_output = btn.render()

	assert "Rendēju elementu" in render_output, "Kļūda renderēšanas loģikā."
	print("Scenārijs 8 darbojas veiksmīgi.")